Firestore/src/Filter.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Firestore;
+
+use Google\Cloud\Firestore\V1\StructuredQuery\CompositeFilter\Operator;
+
+/**
+ * A Query Filter class.
+ *
+ * This class helps the user to create filters for complex queries.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Firestore\FirestoreClient;
+ * use Google\Cloud\Firestore\Filter;
+ *
+ * $firestore = new FirestoreClient();
+ *
+ * $collection = $firestore->collection('users');
+ * $query = $collection->where(Filter::condition('age', '>', 18));
+ * ```
+ */
+class Filter
+{
+    /**
+     * Helper function for `and` filter.
+     *
+     * @param array $filters A filter array.
+     * @return array A composite Filter array.
+     */
+    public static function doAnd(array $filters)
```

Comment:
We changed the name from `or` to `doOr` just to support PHP56, shall we change back to `or` since we are dropping the PHP56 support?

Reply:
We got a go ahead to continue to use the name `and(..)` rather than `doAnd` and skip to the php5.6 tests in workflows.

Reply:
Yeah thats a good idea, let me do that, thank you!

Reply:
Done, Thank you!

Reply:
This PR will not skip the tests in just 5.6, we can manually ignore the failures there for now.